src/d3a/d3a_core/export.py

```diff
@@ -307,7 +314,11 @@ def plot_energy_profile(self, area: Area, subdir: str):
         """
         self.endpoint_buffer.trade_profile.add_sold_bought_lists(
             self.endpoint_buffer.trade_profile.traded_energy_profile)
-        new_subdir = os.path.join(subdir, area.slug)
+        # to do
+        if area.slug != "grid":
+            new_subdir = os.path.join(subdir, area.slug + "-" + area.uuid)
+        else:
+            new_subdir = os.path.join(subdir, area.slug)
```

Comment:
Minor and im not sure whether this is possible, but I wonder whether you can find a smarter way to add the uuid in a more central place and not doing this in every function.